modules/core/src/main/java/org/apache/ignite/internal/processors/security/IgniteSecurityProcessor.java

```diff
@@ -100,7 +105,14 @@ public IgniteSecurityProcessor(GridKernalContext ctx, GridSecurityProcessor secP
     @Override public OperationSecurityContext withContext(SecurityContext secCtx) {
         assert secCtx != null;
 
-        secPrc.touch(secCtx);
+        try {
+            secPrc.touch(secCtx);
+        }
+        catch (SecurityException e) {
+            // Security context is not propagated yet.
+            // Log this warning and proceed an operation with already given security context.
+            log.warning("Failed to check security context [subj=" + secCtx.subject().id() + ", err=" + e + ']');
```

Comment:
Maybe add printing a stack trace for the exception? Not sure if we have utility methods to print the exception stack trace to levels other than ERROR.

I would be OK with either printing the stack trace always as WARNING or conditionally as DEBUG - your choice.

Reply:
I think the following stack trace would be enough:
```
[2020-11-16 12:46:38,417][WARN ][client-connector-#124%security.SecurityThinJavaClientSecurityContextPropagationSelfTest0%][GridEntSecurityProcessor] Failed to wait for security context propagation [subj=567c51fb-8402-490a-ae05-f6a6675f07fb, err=Security context is not ready [subjectId=567c51fb-8402-490a-ae05-f6a6675f07fb, contextPropagationTimeout=3000, sesionInvalidationTimeout=3000, failureDetectionTimeout=10000]]
class org.apache.ignite.plugin.security.SecurityException: Security context is not ready [subjectId=567c51fb-8402-490a-ae05-f6a6675f07fb, contextPropagationTimeout=3000, sesionInvalidationTimeout=3000, failureDetectionTimeout=10000]
	at org.gridgain.grid.internal.processors.security.GridEntSecurityProcessor.waitForSessionContextPropagation(GridEntSecurityProcessor.java:1196)
	at org.gridgain.grid.internal.processors.security.GridEntSecurityProcessor.touch(GridEntSecurityProcessor.java:1002)
	at org.apache.ignite.internal.processors.security.IgniteSecurityProcessor.withContext(IgniteSecurityProcessor.java:108)
	at org.apache.ignite.internal.processors.odbc.ClientListenerNioListener.onMessage(ClientListenerNioListener.java:200)
	at org.apache.ignite.internal.processors.odbc.ClientListenerNioListener.onMessage(ClientListenerNioListener.java:57)
	at org.apache.ignite.internal.util.nio.GridNioFilterChain$TailFilter.onMessageReceived(GridNioFilterChain.java:278)
	at org.apache.ignite.internal.util.nio.GridNioFilterAdapter.proceedMessageReceived(GridNioFilterAdapter.java:108)
	at org.apache.ignite.internal.util.nio.GridNioAsyncNotifyFilter$3.body(GridNioAsyncNotifyFilter.java:135)
	at org.apache.ignite.internal.util.worker.GridWorker.run(GridWorker.java:119)
	at org.apache.ignite.internal.util.worker.GridWorkerPool$1.run(GridWorkerPool.java:69)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: class org.apache.ignite.internal.IgniteFutureTimeoutCheckedException: Timeout was reached before computation completed.
	at org.apache.ignite.internal.util.future.GridFutureAdapter.get0(GridFutureAdapter.java:252)
	at org.apache.ignite.internal.util.future.GridFutureAdapter.get(GridFutureAdapter.java:168)
	at org.apache.ignite.internal.util.future.GridFutureAdapter.get(GridFutureAdapter.java:160)
	at org.gridgain.grid.internal.processors.security.GridEntSecurityProcessor.waitForSessionContextPropagation(GridEntSecurityProcessor.java:1189)
	... 12 more
```